google/cloud/bigtable/examples/bigtable_samples_instance_admin_ext.cc

```diff
@@ -0,0 +1,305 @@
+  auto instances = instance_admin.ListInstances();
+  auto instance_name =
+      instance_admin.project_name() + "/instances/" + instance_id.get();
+  bool instanceExists =
+      instances.end() !=
+      std::find_if(
+          instances.begin(), instances.end(),
+          [&instance_name](google::bigtable::admin::v2::Instance const& i) {
+            return i.name() == instance_name;
+          });
+  // Create instance if does not exists
+  if (not instanceExists) {
+    // [START bigtable_create_dev_instance]
+    std::cout << "\nCreating a DEVELOPMENT Instance: ";
+    google::cloud::bigtable::DisplayName display_name("Put description here");
+
+    // Cluster nodes should not be set while creating Development Instance
+    auto cluster_config = google::cloud::bigtable::ClusterConfig(
+        zone, 0, google::cloud::bigtable::ClusterConfig::HDD);
+    google::cloud::bigtable::InstanceConfig config(
+        google::cloud::bigtable::InstanceId(instance_id), display_name,
+        {{cluster_id.get(), cluster_config}});
+    config.set_type(google::cloud::bigtable::InstanceConfig::DEVELOPMENT);
+
+    auto future = instance_admin.CreateInstance(config).get();
+    std::cout << " Done" << std::endl;
+    // [END bigtable_create_dev_instance]
+  } else {
+    std::cout << "\nInstance " << instance_id.get() << " already exists."
+              << std::endl;
+    return;
+  }
+  std::cout << " Done" << std::endl;
+}
+
+void DeleteInstance(std::string project_id, int argc, char* argv[]) {
+  if (argc != 2) {
+    throw Usage{"delete-instance: <project-id> <instance-id>"};
+  }
+
+  std::string instance_id = ConsumeArg(argc, argv);
+
+  // Create an instance admin endpoint.
+  google::cloud::bigtable::InstanceAdmin instance_admin(
+      google::cloud::bigtable::CreateDefaultInstanceAdminClient(
+          project_id, google::cloud::bigtable::ClientOptions()));
+
+  // [START bigtable_delete_instance]
+  std::cout << "\nDeleting Instance: ";
+  instance_admin.DeleteInstance(instance_id);
+  std::cout << " Done" << std::endl;
+  // [END bigtable_delete_instance]
+}
+
+void CreateCluster(std::string project_id, int argc, char* argv[]) {
+  if (argc != 4) {
+    throw Usage{
+        "create-cluster: <project-id> <instance-id> <cluster-id> <zone>"};
+  }
+
+  google::cloud::bigtable::InstanceId instance_id(ConsumeArg(argc, argv));
+  google::cloud::bigtable::ClusterId cluster_id(ConsumeArg(argc, argv));
+  std::string const zone = ConsumeArg(argc, argv);
+
+  // Create an instance admin endpoint.
+  google::cloud::bigtable::InstanceAdmin instance_admin(
+      google::cloud::bigtable::CreateDefaultInstanceAdminClient(
+          project_id, google::cloud::bigtable::ClientOptions()));
+
+  std::cout << "\nCheck Instance exists: " << std::endl;
+  auto instances = instance_admin.ListInstances();
+  auto instance_name =
+      instance_admin.project_name() + "/instances/" + instance_id.get();
+  bool instanceExists =
```

Comment:
Please follow the C++ naming conventions here (and everywhere).

Reply:
fixed.